spec/database_admin_spec.rb

```diff
@@ -67,48 +67,175 @@
         expect(subject).to receive(:ask_local_file_options).once
         say ""
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::LOCAL_FILE)
+        expect(subject.backup_type).to eq("local")
       end
 
       it "calls #ask_local_file_options when choosen" do
         expect(subject).to receive(:ask_local_file_options).once
         say "1"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::LOCAL_FILE)
+        expect(subject.backup_type).to eq("local")
       end
 
       it "calls #ask_nfs_file_options when choosen" do
         expect(subject).to receive(:ask_nfs_file_options).once
         say "2"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::NFS_FILE)
+        expect(subject.backup_type).to eq("nfs")
       end
 
       it "calls #ask_smb_file_options when choosen" do
         expect(subject).to receive(:ask_smb_file_options).once
         say "3"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::SMB_FILE)
+        expect(subject.backup_type).to eq("smb")
       end
 
       it "calls #ask_s3_file_options when choosen" do
         expect(subject).to receive(:ask_s3_file_options).once
         say "4"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::S3_FILE)
+        expect(subject.backup_type).to eq("s3")
       end
 
       it "calls #ask_ftp_file_options when choosen" do
         expect(subject).to receive(:ask_ftp_file_options).once
         say "5"
         subject.ask_file_location
-        expect(subject.backup_type).to eq(described_class::FTP_FILE)
+        expect(subject.backup_type).to eq("ftp")
       end
 
       it "cancels when CANCEL option is choosen" do
         say "6"
         expect { subject.ask_file_location }.to raise_error signal_error
       end
+
+      # this is the complete implementation. the other 2 are paired down version of this
+      context "with localized file upload" do
```

Comment:
_**Pedantic**_:  maybe "with I18n custom menu configs"?

If changed, make sure to change in the other two spots.